core/src/main/resources/org/apache/spark/ui/static/historypage.js

```diff
@@ -37,6 +37,22 @@ function formatDuration(milliseconds) {
   return hours.toFixed(1) + " h";
 }
 
+function makeIdNumeric(id) {
+  var strs = id.split("-");
+  if (strs.length < 3) {
+    return id;
+  }
+  var appSeqNum = strs[2];
+  var resl = strs[0] + "-" + strs[1] + "-";
+  var diff = 10 - appSeqNum.length;
+  while (diff > 0) {
```

Comment:
we should at least comment this.  If java script has a function to pad or print strings it certain length it would be better.